packages/asset-swapper/test/utils/rfqt_mocker.ts

```diff
@@ -0,0 +1,65 @@
+import { SignedOrder } from '@0x/order-utils';
+import { StatusCodes } from '@0x/types';
+import axios from 'axios';
+import AxiosMockAdapter from 'axios-mock-adapter';
+
+/**
+ * Ensures that we throw if a HTTP request is made
+ * that is not one of the known mocked routes
+ * See https://github.com/ctimmerm/axios-mock-adapter/issues/173
+ */
+const setupAdapterToThrowOnUnknownRequest = (mockAdapter: AxiosMockAdapter) => {
+    mockAdapter.onAny().reply(config => {
+        throw new Error(
+            `Could not find mock for ${config.url} ${JSON.stringify(config.params)} ${JSON.stringify(config.headers)}`,
+        );
+    });
+};
+
+interface RfqtMakerParams {
+    sellToken: string;
+    buyToken: string;
+    sellAmount: string;
+    takerAddress: string;
+}
+export interface MockedSuccessfulRfqtRequest {
+    endpoint: string;
+    expectedApiKey: string;
+    expectedParams: RfqtMakerParams;
+    response: SignedOrder;
+}
+export const rfqtMocker = {
+    withMockedRfqtRequests: async (
```

Comment:
Instead of creating all this abstraction, perhaps I should let the consumer of this function mock any arbitrary URL, param and header combo with any arbitrary data